spec/lib/facilities/gis_client_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Facilities::GisClient do
+  RSpec::Matchers.define :has_order_and_offset do |order, offset|
+    match do |params|
+      params[:orderByFields].eql?(order) && params[:resultOffset].eql?(offset)
+    end
+  end
+
+  let(:faraday_response_offset_0) { double(Faraday::Response.new) }
+  let(:faraday_response_offset_10) { double(Faraday::Response.new) }
+  let(:faraday_response_offset_20) { double(Faraday::Response.new) }
+
+  describe 'get_all_facilities' do
+    subject { described_class.new }
+
+    before(:each) do
+      allow_any_instance_of(
+        Faraday::Connection
+      ).to receive(:get).with(anything, has_order_and_offset('field', 0)).and_return(faraday_response_offset_0)
+
+      allow_any_instance_of(
+        Faraday::Connection
+      ).to receive(:get).with(anything, has_order_and_offset('field', 10)).and_return(faraday_response_offset_10)
+
+      allow_any_instance_of(
+        Faraday::Connection
+      ).to receive(:get).with(anything, has_order_and_offset('field', 20)).and_return(faraday_response_offset_20)
+    end
+
+    it 'passes the correct offset to the query when looping twice' do
+      allow(faraday_response_offset_0).to receive(:env).and_return(double(body: [*1..12]))
```

Comment:
You're passing a `max_record_count` of 10, but then mocking that the response returns 12 results.  
Is that a possibility?

Reply:
I believe so as the GIS_Client only passes an offset to the server to make a request and does not pass a maximum number of records to retrieve.  The `max_record_count` only comes into play when looking at a particular response from the GIS server and at that point is only looking at the length of the body.

Reply:
If we are assuming the server returns "maxRecordCount" and it should be the "per page" variable, then we should also assume the first expectation would return 10 records. We can investigate the server's behavior more in depth later.